truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/impl/Accessor.java

```diff
@@ -420,6 +422,29 @@ protected String toString(TruffleLanguage<?> language, Env env, Object obj) {
         return API.toString(language, env, obj);
     }
 
+    protected boolean supportsOnLoopCount() {
+        return false;
+    }
+
+    protected void onLoopCount(Node source, int count) {
+        // optimized calltarget is not existent on default runtimes
+        if (OPTIMIZEDCALLTARGET != null) {
```

Comment:
I am not able to find out who provides OPTIMIZEDCALLTARGET instance, but I am starting to get scared that it is someone outside of Truffle repository. That isn't good idea as that makes Accessor part of API and we would have to evolve it compatibly to prevent linkage errors.

Using Accessor pattern for the loopcount reporting is good idea, just the way to communicate this with runtime as to be different.